src/libraries/System.Runtime/tests/System/NullableTests.cs

```diff
@@ -93,6 +94,48 @@ public static void GetUnderlyingType_NullType_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("nullableType", () => Nullable.GetUnderlyingType((Type)null));
         }
 
+        [Fact]
+        public static void GetValueRefOrDefaultRef_WithValue()
+        {
+            static void Test<T>(T before, T after)
+                where T : struct
+            {
+                T? nullable = before;
+                ref readonly T reference = ref Nullable.GetValueRefOrDefaultRef(nullable);
+
+                Assert.Equal(nullable!.Value, before);
+
+                Unsafe.AsRef<T>(reference) = after;
+
+                Assert.Equal(nullable.Value, after);
+            }
```

Comment:
Can you add a test variant of this that does the same thing, except with a null T?, validating that after writing unsafely to the value, Value still throws an exception and HasValue is still false, but that you can still read the value out via the ref readonly?

Reply:
Sure thing! Done in https://github.com/dotnet/runtime/pull/64677/commits/f54b167f61a3497fd63b3d70e59bc2d483e0c07a 👍